internal/resolution/manifest/maven.go

```diff
@@ -843,7 +854,6 @@ func updateDependency(w io.Writer, enc *xml.Encoder, raw string, patches map[Mav
 					// A Maven dependency key consists of Type and Classifier together with GroupID and ArtifactID.
 					if patch.DependencyKey == rawDep.Key() {
 						req = patch.NewRequire
-						patches[patch] = true
```

Comment:
this is no longer needed since we pre-compute which patches are to be added in `buildPatches`.